### PR TITLE
Add support layout

### PR DESCRIPTION
**Pull request by jaroel**
- Support setting/modifying 'layout' on DX and AT content endpoints
- Support getting the defined layouts on the root types endpoint

### CODE REVIEW

File: test-no-uncommitted-doc-changes.in [hunk 1]
```diff
@@ -20,7 +20,7 @@ if [ "$PLONE_VERSION" = "5.0.x" ]; then
     exit 0
 fi
 
-changes=$(git diff --exit-code $DUMPS_DIR)
+changes=$(git diff --ignore-space-at-eol --exit-code $DUMPS_DIR)
```
Comment (tisto) [inline test-no-uncommitted-doc-changes.in]: @jaroel nice one. Though, I really wonder why the auto-generated json files have blanks at the end of the line in the frist place. cc @lukasgraf

Comment (lukasgraf) [inline test-no-uncommitted-doc-changes.in]: @tisto the spaces at the end of the line are an [artifact of pretty printing JSON using the Python stdlib `json` module](https://bugs.python.org/issue16333) a certain way. Using `separators=(',', ': ')` should avoid trailing whitespace.

When dumping the request body, we [pretty print the JSON directly in `test_documentation.py`](https://github.com/plone/plone.restapi/blob/c6f2eb3572f12e86391c841fbdf47440e988a64f/src/plone/restapi/tests/test_documentation.py#L50-L51) when saving the request. When saving responses however, we save them just like `plone.restapi` returns them, which [currently adds trailing whitespace](https://github.com/plone/plone.restapi/blob/c6f2eb3572f12e86391c841fbdf47440e988a64f/src/plone/restapi/services/__init__.py#L22).

What still is a mystery to me is how this behavior could be flaky. I basically suspect to possible culprits: Either text editors that remove trailing whitespace on save, or a git config (`apply.whitespace` possibly?) that does that.

Either way, we don't want those details tripping up that test, so `+1` on @jaroel introducing `--ignore-space-at-eol` 👍

Comment (jaroel) [inline test-no-uncommitted-doc-changes.in]: Yep, my editor removes whitespace at EOL. I think I'm not the only one, but don't down on that ;)

Comment (tisto) [inline test-no-uncommitted-doc-changes.in]: My editor removes whitespace at EOL as well. Though, I usually do not touch the auto-generated files at all. :)

**Pull request merged**